packages/components/src/List/ListComposition/Manager/useDisplayMode.js

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+export default function useDisplayMode(initialDisplayMode, onChange) {
+	const [displayMode, setDisplayMode] = useState(initialDisplayMode);
+
+	function onDisplayModeChange(event, value) {
+		if (onChange) {
+			onChange(event, value);
+		} else {
+			setDisplayMode(value);
+		}
```

Comment:
What about this :
```suggestion
export default function useDisplayMode(initialDisplayMode, onChange, overrideDisplayMode){
	const [displayMode, setDisplayMode] = useState(initialDisplayMode);
	if (overrideDisplayMode) {
	     setDisplayMode(overrideDisplayMode)
	}

	function onDisplayModeChange(event, value) {
		setDisplayMode(value);
		if (onChange) {
			onChange(event, value);
		}
	}
	return [displayMode, onDisplayModeChange];
}

```
Just an idea I didn't test it, have to check what happens in render cycle. But with this solution we are making our hook the unique source of truth and with ovveriding possibility.

Reply:
Not sure it's a good idea.
But that will collide with the previous comment subject. Let's try something for the previous one first